enterprise/micronaut/src/org/netbeans/modules/micronaut/maven/BuildActions.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.micronaut.maven;
+
+import java.util.Arrays;
+import javax.swing.Action;
+import org.netbeans.api.project.Project;
+import org.netbeans.modules.maven.api.NbMavenProject;
+import org.netbeans.spi.project.ActionProvider;
+import org.netbeans.spi.project.ui.support.ProjectActionPerformer;
+import org.netbeans.spi.project.ui.support.ProjectSensitiveActions;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.Lookup;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author sdedic
+ */
+public class BuildActions {
+    
+    @ActionID(id = "org.netbeans.modules.micronaut.maven.nativeBuild", category = "Project")
+    @ActionRegistration(displayName = "#LBL_NativeBuild", lazy=false)
+    @ActionReference(position = 520, path = "Projects/org-netbeans-modules-maven/Actions")
+    @NbBundle.Messages({
+        "LBL_NativeBuild=Build Native Executable"
+    })
+    public static Action    createNativeBuild() {
+        return ProjectSensitiveActions.projectSensitiveAction(new NativeBuildPerformer(), Bundle.LBL_NativeBuild(), null);
+    }
+    
+    private static class NativeBuildPerformer implements ProjectActionPerformer {
+        @Override
+        public boolean enable(Project project) {
+            NbMavenProject nbmp = project.getLookup().lookup(NbMavenProject.class);
+            if (nbmp == null) {
+                return false;
+            }
+            ActionProvider ap = project.getLookup().lookup(ActionProvider.class);
+            return 
+                    nbmp.getMavenProject().getPlugin("io.micronaut.build:micronaut-maven-plugin") != null &&
```

Comment:
I was hoping this (presence of some plugin) could be specified declaratively. Isn't there already a support for a plugin to trigger some configuration and configurations adding some actions?

Reply:
Hmmm, good catch, will try to use plain `isActionEnabled`.

Reply:
Addressed in 2917d2be433e